pkg/kubelet/kubelet.go

```diff
@@ -697,8 +702,9 @@ func makeBinds(container *api.Container, podVolumes volumeMap) (binds []string)
 func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Container, netMode, ipcMode string) (*kubecontainer.RunContainerOptions, error) {
 	var err error
 	opts := &kubecontainer.RunContainerOptions{
-		NetMode: netMode,
-		IpcMode: ipcMode,
+		NetMode:    netMode,
+		IpcMode:    ipcMode,
+		CgroupRoot: kl.cgroupRoot,
```

Comment:
Yes, let's do cgroups for multiple QoS tiers and for pods in subsequent PRs. I don't think we need much more discussion on them, but we will need an (additive) API change, to add resources at the pod level. 

Given where we are, I don't think we need to be that rigorous about changing the hierarchy. If we evacuate the node when we update the kubelet, which we need to be able to do for kernel upgrades, we'll be fine.

As for Kubelet, we're working on making it run as a container. Eventually I'd like to even run it as a pod of the highest QoS tier, but putting it into a custom cgroup root seems fine for now.